classes/class-ledyer-api.php

```diff
@@ -34,10 +34,26 @@ public function get_order_session( $order_id ) {
 	 *
 	 * @return mixed|\WP_Error
 	 */
-	public function create_order_session( $data ) {
+	public function create_order_session( $data, $order_id = false ) {
+		if ( $order_id ) {
+			$order = wc_get_order( $order_id );
+			if ( $order ) {
+				$data['settings']['urls']['confirmation'] = $order->get_checkout_order_received_url();
```

Comment:
I suggest moving this control to [`Merchant_URLs::get_confirmation_url`](https://github.com/krokedil/ledyer-checkout-for-woocommerce/blob/develop-hpp-redirect/classes/class-ledyer-merchant-urls.php#L94-L113). You'll also need this control:

```php
is_wc_endpoint_url( 'order-pay' ) || 'redirect' === ( $this->settings['checkout_flow'] ?? 'embedded' )
```

The order ID can then be retrieved directly from the order-pay endpoint:

```php
if ( is_wc_endpoint_url( 'order-pay' ) ) {
	$order_id = absint( get_query_var( 'order-pay', 0 ) );
	$order    = wc_get_order( $order_id );
}
```

However, this only work for order-pay, and since we need to support the redirect flow, we need to make sure this is the order ID is passed in the `process_payment` function when calling `Create_HPP`.